File: Sort/src/SelectionSort.java
import java.util.ArrayList;

public class SelectionSort
{
    public static int CompareCount = 0;

    public static void Sort(ArrayList<Integer> list)
    {
        CompareCount = 0;

        int maxIndex, temp;

        for (int k = list.size() - 1; k > 0; k--)
        {
            maxIndex = 0;
            for (int index = 0; index <= k; index++)
            {
                if (list.get(maxIndex) < list.get(index))
                {
                    maxIndex = index;
                    CompareCount++;
                }
            }

            temp = list.get(k);
            list.set(k, list.get(maxIndex));
            list.set(maxIndex, temp);
        }
    }
}

File: Algorithm/SelectionSort.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Algorithm
{
    public class SelectionSort : ISortingAlgorithm
    {
        public void Sort(int[] array)
        {
            for (int last = array.Length - 1; last > 0; last--)
            {
                int largestIndex = FindLargestIndex(array, last);

                if (largestIndex != last)
                {
                    Swap(array, largestIndex, last);
                }
            }
        }

        private int FindLargestIndex(int[] array, int range)
        {
            int largestIndex = 0;

            for (int index = 1; index <= range; index++)
            {
                if (array[index] > array[largestIndex])
                {
                    largestIndex = index;
                }
            }

            return largestIndex;
        }

        private void Swap(int[] array, int x, int y)
        {
            int temp = array[x];
            array[x] = array[y];
            array[y] = temp;
        }
    }
}
